src/HTTP/Factory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/* Copyright (c) 2016 Fabian Schmid <fs@studer-raimann.ch> Extended GPL, see docs/LICENSE */
+namespace ILIAS\HTTP;
+
+use ILIAS\HTTP\Cookies\CookieJar;
+use ILIAS\HTTP\Response\rendering\RenderingException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Interface Factory
```

Comment:
Hi @Amstutz 
I am sorry, I forgot to push the implementation of the factory. I will do that as fast as possible.

**clone vs constructor**
I was not aware of a performance benefit of the clone method. Is there a "considerable" performance difference between the two methods? If thats the case, I will change the implementation.